### DISCUSSION TITLE
How to disable NGINX access logs in Docker container logs

### DISCUSSION THREAD

**Discussion by zionsg** [serversideup/docker-php#94] [Q&A] [answered]
Using `serversideup/php:8.1-fpm-nginx` as base image in my Dockerfile currently. I currently output my application logs to the Docker container logs via `php://stdout`. How do I turn off the access logs that is showing in the Docker container logs to reduce clutter?

Sample logs (the ones I want to get rid of are those like the last log starting with an IP address):
```
app    | [11-Nov-2022 06:38:48] NOTICE: fpm is running, pid 76
app    | [11-Nov-2022 06:38:48] NOTICE: ready to handle connections
app    | [11-Nov-2022 06:38:48] NOTICE: systemd monitor interval set to 10000ms
app    | Sample log output from my PHP application via php://stdout
app    | 172.29.0.1 - - [11/Nov/2022:06:43:56 +0000] "GET / HTTP/1.1" 200 140 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36"
```

I have tried modifying the PHP-FPM and Nginx config in the Dockerfile but to no avail. Example:
```
FROM serversideup/php:8.1-fpm-nginx

RUN echo "access.log = /dev/null" >> /etc/php/current_version/fpm/pool.d/www.conf
RUN sed 's/\(access\|error\)_log .*/\1_log \/dev\/null;/' /etc/nginx/nginx.conf

# Tried to restart the services as well but the commands would fail
# RUN /package/admin/s6/command/s6-svc -r php-fpm
# RUN service php8.1-fpm restart

# No need for ENTRYPOINT or CMD at the end as the base image will automatically start up the Nginx web server
# and expose the standard ports 80/443?
```

**Comment by zionsg** [comment, marked as answer]
Realized my mistake - missed out the `-i` option for `sed` which updates the Nginx configuration file with the new text.

For those who have come here, here's the contents of my Dockerfile for future reference
```
##
# Dockerfile
##

# Base image - see https://github.com/serversideup/docker-php
FROM serversideup/php:8.1-fpm-nginx

# Disable PHP-FPM and Nginx logs so that they will not clutter up Docker
# container logs and make it hard to sift out the application logs which are
# also output to Docker container logs via stdout
RUN echo "access.log = /dev/null" >> /etc/php/current_version/fpm/pool.d/www.conf
RUN sed --in-place 's/\(access\|error\)_log .*/\1_log \/dev\/null;/' /etc/nginx/nginx.conf

# Install additional system packages not found in base image
RUN apt-get update \
    && apt-get install --yes --no-install-recommends php8.1-imap vim \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/*

# Enable additional PHP extensions
RUN phpenmod imap

# Create subfolders for application (/var/www/html created in base image)
RUN mkdir -p /var/www/html/public \
    && mkdir -p /var/www/html/src \
    && mkdir -p /var/www/html/tmp
WORKDIR /var/www/html

# Copy only essential files and folders - Docker recommends using COPY instruction over ADD
# Placing the copy commands explicitly here is easier to troubleshoot
# than using .dockerignore. Do NOT copy .env inside here, use docker-compose.yml
# or Docker CLI to set environment variables for the container instead.
COPY public/ /var/www/html/public/
COPY src/ /var/www/html/src/
COPY composer.* /var/www/html/

# Install application dependencies
RUN composer install

# Command for VERSION.txt placed last as it always changes, making Docker always rebuild this layer and subsequent ones
COPY VERSION.txt /var/www/html/

# No need for ENTRYPOINT or CMD at the end of this Dockerfile as the base image will automatically
# start up the Nginx web server and expose the standard ports 80/443
```

**Reply by jaydrogers** [reply]
Thanks for publishing your solution 🙌

**Reply by TRONJon** [reply]
@jaydrogers This took me several days of searching to find this solution. 
For anyone using docker-php in fairly busy products, the docker logs become swamped in access log output in seconds.

Zionsg's solution involves surgically modifying the nginx config during the build stage.
Do you think there could be any chance of this being in the official documentation? I ready every page of it with the hopes of finding something talking about access log output but found nothing.

Is it something which an environment variable could toggle?

What's your thoughts on this - I'd like to save the time of other developers in the same situation

Jon

**Reply by jaydrogers** [reply]
Can you help me out and open a detailed feature request?

https://github.com/serversideup/docker-php/discussions/66

I could see making a `PHP_ACCESS_LOG` and `NGINX_ACCESS_LOG` default it to `/dev/stdout`

**Reply by snoek09** [reply]
Just bumped into the same issue and I didn't see a feature request for this yet.
Added it here: https://github.com/serversideup/docker-php/discussions/508

**Comment by robsontenorio** [comment]
Submitted #534
